unity/Assets/Scripts/CanOpen_Object.cs

```diff
@@ -97,529 +78,218 @@ void Start ()
-        if (!canReset)
-		{
-			//Interact();
-
-            //print("we are calling Reset() now");
-            //we are still open, trying to close, but hit something - reset to open
-			if(isOpen)
-			{
-				for (int i = 0; i < MovingParts.Length; i++)
-                {
-                    if (movementType == MovementType.Rotate)
-                    {
-						iTween.RotateTo(MovingParts[i], iTween.Hash(
-                        "rotation", openPositions[i] * currentOpenPercentage,
-                        "islocal", true,
-                        "time", animationTime,
-                        "easetype", "linear"));
-
-                    }
-
-
-                    else if (movementType == MovementType.Slide)
-                    {
-						iTween.MoveTo(MovingParts[i], iTween.Hash(
-                        "position", openPositions[i] * currentOpenPercentage,
-                        "islocal", true,
-                        "time", animationTime,
-                        "easetype", "linear"));
-
-                    }
-
-                    else if (movementType == MovementType.ScaleY)
-                    {
-                        iTween.ScaleTo(MovingParts[i], iTween.Hash(
-                        "scale", new Vector3(openPositions[i].x, closedPositions[i].y + (openPositions[i].y - closedPositions[i].y) * currentOpenPercentage, openPositions[i].z),
-                        "islocal", true,
-                        "time", animationTime,
-                        "easetype", "linear"));
-                    }
-
-                }
-			}
-
-			else
-			{
-				for (int i = 0; i < MovingParts.Length; i++)
-                {
-                    if (movementType == MovementType.Rotate)
-                    {
-						iTween.RotateTo(MovingParts[i], iTween.Hash(
-                        "rotation", closedPositions[i],
-                        "islocal", true,
-                        "time", animationTime,
-                        "easetype", "linear"));
-
-                    }
-
-
-                    else if (movementType == MovementType.Slide)
-                    {
-						iTween.MoveTo(MovingParts[i], iTween.Hash(
-                        "position", closedPositions[i],
-                        "islocal", true,
-                        "time", animationTime,
-                        "easetype", "linear"));
-                    }
-
-
-                    else if(movementType == MovementType.ScaleY)
-                    {
-                        iTween.ScaleTo(MovingParts[i], iTween.Hash(
-                        "scale", closedPositions[i],
-                        "islocal", true,
-                        "time", animationTime,
-                        "easetype", "linear"));
-                    }
-                }
-			}
-
+    // note: reset can interrupt the Interact() itween call because
+    // it will start a new set of tweens before onComplete is called from Interact()... it seems
+    public void Reset() {
```

Comment:
`Reset`, which is called within `OnTriggerEnter` used to also treat Open as a Toggle, where if the object is open, let's close it, and if the object is closed, let's open it. It duplicated much of the functionality as `Interact` (which animates the opening of the objects).

Now, on failure, we just call Interact, passing in the `startOpenness` before the failed action started.

Reply:
I think there is potential to change this behavior to instead reset to the `startOpenness` we can have an option to simply stop the animation wherever it was when `Reset` is called.

I don't think this change should be in this PR, but I can see this sort of functionality being useful when we eventually implement open/close with the Mid Level Arm. I think the behavior in that case when an arm tries to open something is to "open" it via the animation, but if the arm hits something and can no longer follow through with the opening motion, the openable object stops the animation.